File: backend/src/main/java/com/muk/food/FoodService.java
package com.muk.food;

import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class FoodService {

    private final FoodRepository repository;

    public FoodService(FoodRepository repository) {
        this.repository = repository;
    }

    public List<Producto> getAllFoods() {
        return repository.findAll();
    }

    public Optional<Producto> getFoodById(String id) {
        if (id == null || id.trim().isEmpty()) return Optional.empty();
        return repository.findById(id.trim());
    }

    public List<Producto> getFoodsByCategory(String category) {
        if (category == null || category.trim().isEmpty()) return List.of();
        return repository.findByCategory(category.trim());
    }

    public List<Producto> searchFoods(String query) {
        if (query == null || query.trim().isEmpty()) return List.of();
        return repository.searchByName(query.trim());
    }
}

File: src/services/FoodService.js
// Servicio: validación y delegación al repositorio.

function requireNonEmptyString(value, fieldName) {
  if (typeof value !== 'string' || value.trim().length === 0) {
    throw new Error(`${fieldName} must be a non-empty string`);
  }
  return value.trim();
}

export class FoodService {
  constructor({ repository }) {
    this.repository = repository;
  }

  getAllFoods() {
    return this.repository.findAll();
  }

  getFoodById(id) {
    const safeId = requireNonEmptyString(id, 'id');
    return this.repository.findById(safeId);
  }

  getFoodsByCategory(category) {
    const safeCategory = requireNonEmptyString(category, 'category');
    return this.repository.findByCategory(safeCategory);
  }

  searchFoods(query) {
    if (typeof query !== 'string' || query.trim().length === 0) return [];
    return this.repository.searchByName(query);
  }
}

